src/web/common/components/chat/chat.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Channel,
+  ChannelHeader,
+  Chat as ChatProvider,
+  MessageInput,
+  MessageList,
+  Thread,
+  Window,
+} from 'stream-chat-react'
+import 'stream-chat-react/dist/css/v2/index.css'
+import { useEffect, useMemo, useRef, useState } from 'react'
+
+import { useAuth } from '@/web/auth'
+import { type DefaultEmptyType } from '@/common'
+
+import {
+  type ChatChannel,
+  useChatClient,
+  type UserConnection,
+} from './hooks/use-chat-client'
+import { createChatUserCredentials } from './library/create-chat-user-credentials'
+
+type ChatProps = {
+  channelId: string
+}
+
+export const Chat = ({ channelId }: ChatProps) => {
+  const { data: user } = useAuth()
+
+  const { chatClient, connectUser, joinChannel } = useChatClient()
+  const [channel, setChannel] = useState<DefaultEmptyType<ChatChannel>>()
+  const connectPromiseRef =
+    useRef<DefaultEmptyType<Promise<UserConnection | void>>>()
+
+  const chatUser = useMemo(
+    () => (user ? createChatUserCredentials(user) : null),
+    [user]
+  )
+
+  useEffect(() => {
+    if (!chatUser || connectPromiseRef.current) return
+
+    connectPromiseRef.current = connectUser(chatUser).then(() =>
+      setChannel(
+        joinChannel(chatUser, {
+          channelId,
+          title: 'Talk about React',
+        })
+      )
+    )
+  }, [channelId, chatUser, connectUser, joinChannel])
```

Comment:
1. Think how to avoid the `connectPromiseRef` definition.
2. Think how to simplify the promise chain `connectUser -> setChannel -> joinChannel`. Maybe use new version of React.
3. Think how to remove the `channel` state.